### PR TITLE
Fix `Cache<_, Value, _>` type parameter variance (covariant -> invariant)

### PR DESCRIPTION
**Pull request by KhraksMamtsov**
## Type

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Related

- Related Issue #
- Closes #

### CODE REVIEW

**Comment by mikearnaldi** [comment]
just changing `out` to `in out` doesn't actually affect variance when compared structurally, we should use the `Variance` trick we use in `Effect` and on the other types to force a specific variance

**Comment by KhraksMamtsov** [comment]
Done.
I used the `Cache.Variance` type for the regular `Cache`, and for `ConsumerCache` - `Cache.ConsumerVariance`.
And now `CacheImpl` contains both `Variance` symbols.
I was looking at `Queue` which implements `Enqueue` + `Dequeue`

File: packages/effect/src/Cache.ts [hunk 1]
```diff
@@ -22,6 +22,18 @@ export const CacheTypeId: unique symbol = internal.CacheTypeId
  */
 export type CacheTypeId = typeof CacheTypeId
 
+/**
+ * @since 2.0.0
```
Comment (tim-smart) [inline packages/effect/src/Cache.ts:26]:
```suggestion
 * @since 3.6.2
```

**Review by tim-smart — CHANGES_REQUESTED** [review]

File: packages/effect/src/Cache.ts (hunk 1 above)
Comment (KhraksMamtsov) [reply packages/effect/src/Cache.ts]: Done

File: packages/effect/src/Cache.ts [hunk 2]
```diff
@@ -22,6 +22,18 @@ export const CacheTypeId: unique symbol = internal.CacheTypeId
  */
 export type CacheTypeId = typeof CacheTypeId
 
+/**
+ * @since 3.6.2
```
Comment (tim-smart) [inline packages/effect/src/Cache.ts:26]:
```suggestion
 * @since 3.6.4
```

File: packages/effect/src/Cache.ts [hunk 3]
```diff
@@ -22,6 +22,18 @@ export const CacheTypeId: unique symbol = internal.CacheTypeId
...
+ * @category symbols
+ */
+export const ConsumerCacheTypeId: unique symbol = internal.ConsumerCacheTypeId
+
+/**
+ * @since 3.6.2
```
Comment (tim-smart) [inline packages/effect/src/Cache.ts:32]:
```suggestion
 * @since 3.6.4
```

File: packages/effect/src/Cache.ts [hunk 4]
```diff
@@ -156,8 +172,19 @@ export declare namespace Cache {
...
-  export interface Variance<in out Key, out Value, out Error> {
+  export interface Variance<in out Key, in out Value, out Error> {
     readonly [CacheTypeId]: {
+      readonly _Key: Types.Invariant<Key>
+      readonly _Error: Types.Covariant<Error>
+      readonly _Value: Types.Invariant<Value>
+    }
+  }
+  /**
+   * @since 3.6.2
```
Comment (tim-smart) [inline packages/effect/src/Cache.ts:183]:
```suggestion
   * @since 3.6.4
```

**Review by tim-smart — CHANGES_REQUESTED** [review]

**Review by tim-smart — APPROVED** [review]

**Pull request merged**